### DISCUSSION TITLE
NextJS 13 + MongoDB - Data won't update in real time in production, but works on dev

### DISCUSSION THREAD

**Discussion by melodyxpot** [vercel/next.js#57041] [App Router] [answered]
Hi everyone.
I am trying to implement CRUD feature in NextJS app router with MongoDB.
But in production mode, I added one data and tried to get full data of that but added new data doesn't appear.
This is my api code to get full data
```
import Prompts from "@/models/Prompt";
import { connectDb } from "@/utils/mongodb";
import { NextResponse } from "next/server";

/**
 * @method GET
 * @returns NextResponse
 * @description Find All Prompts and return
 */
export const GET = async () => {
	await connectDb();

	try {
		const result = await Prompts.find();
		return NextResponse.json(result.reverse(), { status: 200 });
	} catch (err) {
		return NextResponse.json("Failed to get Prompt Template.", {
			status: 500
		});
	}
};
```
The added data is saved in mongodb correctly and I can see that data in the MongoDB Compass.
But backend api doesn't pull only that data.
In development mode, all things are working well but not working on production mode.
I tried to google but other developer is facing this problem.
https://stackoverflow.com/questions/76787615/nextjs-13-w-mongodb-data-wont-update-in-real-time-in-production-but-works-o
What is the solution for this?

**Comment by icyJoseph** [comment, marked as answer]
Your GET function might be getting statically optimised.

Try adding `export const dynamic = 'force-dynamic'` from the file.

- https://nextjs.org/docs/app/building-your-application/routing/route-handlers#caching

**Reply by YashKaliaS** [reply]
Thank you workss!!

**Reply by ChrisCoder9000** [reply]
Thank you. You deserve a hug

**Comment by melodyxpot** [comment]
Thank you. Your answer is always helpful.
